core/shared/src/main/scala-2.13+/ChunkBuilder.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.collection.mutable.{ ArrayBuilder, Builder }
+
+/**
+ * A `ChunkBuilder[A]` can build a `Chunk[A]` given elements of type `A`.
+ * `ChunkBuilder` is a mutable data structure that is implemented purely for
+ * compatibility with Scala's collection library and should not be used for
+ * other purposes.
+ *
+ * Its implementation wraps an `ArrayBuilder`, delegating all operations to
+ * it. Because we need a `ClassTag` to construct an `Array` but cannot require
+ * one, the implementation defers the creation of the underlying
+ * `ArrayBuilder` until the first element is added, using the `ClassTag` of
+ * that element as the `ClassTag` of the `ArrayBuilder`. This is similar to
+ * the approach in `Chunk`.
+ */
+private[zio] trait ChunkBuilder[A] extends Builder[A, Chunk[A]]
+
+private[zio] object ChunkBuilder {
+
+  /**
+   * Constructs a new `ChunkBuilder`.
+   */
+  def make[A]: ChunkBuilder[A] =
```

Comment:
```suggestion
  def make[A](capacity: Int = 10): ChunkBuilder[A] =
```

I saw you could call `ensureSize` on `ArrayBuilder` if you sub class it. Which would allow "pre-allocation" in cases where it's known about how many things will be added.

Reply:
Yes, I think we want to override the `sizeHint` method. Because these methods will mostly be called by Scala collection library combinators that are outside our control.